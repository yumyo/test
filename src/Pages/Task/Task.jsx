import React from "react";

const Task = (props) => {
  const rewards = props.rewards.allRewards;
  const listItems = rewards.map((reward) => (
    <div key={reward.id} className="reward-card">
      <img src={reward.image} alt={reward.name} />
      <div className="reward-card__content">
        <h2 className="h4">{reward.name}</h2>
        <p>{reward.description}</p>
        <p className="reward-card__data">
          <span>
            Points: <strong>{reward.points}</strong>
          </span>
          <span>
            Cap: <strong>{reward.capPerAmbassador}</strong>
          </span>
          <span>
            Quantity: <strong>{reward.quantity}</strong>
          </span>
        </p>
      </div>
    </div>
  ));
  return <div className="rewards">{listItems}</div>;
};

Task.propTypes = {};

export default Task;
