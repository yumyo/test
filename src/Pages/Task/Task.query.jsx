import { graphql } from "react-apollo";
import { flowRight as compose } from "lodash";
import gql from "graphql-tag";
import { useQuery } from "@apollo/react-hooks";

const getAllRewards = gql`
  query allRewards($page: Int!, $filter: RewardFilter) {
    allRewards(page: $page, filter: $filter) {
      id
      name
      description
      image
      quantity
      points
      capPerAmbassador
    }
  }
`;
export default getAllRewards;
