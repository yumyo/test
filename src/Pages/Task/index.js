import React, { useState } from "react";
import { Query } from "react-apollo";
import getAllRewards from "./Task.query.jsx";
import Task from "./Task.jsx";
import Select from "react-select";

function MyTask() {
  const [value, setValue] = useState({ campaign_id: "001" });

  return (
    <Query
      query={getAllRewards}
      variables={{
        page: 0,
        filter: value,
      }}
    >
      {({ loading, error, data, refetch }) => {
        if (loading) return <p>Loadingâ€¦</p>;
        if (error) return <p>Error :(</p>;
        const handleSelect = (e) => {
          let filterValue = e.value;
          let changeFilter = {
            campaign_id: "001",
            points: filterValue,
          };
          refetch({ variables: { page: 0, filter: changeFilter } });
          setValue(changeFilter);
        };

        const rewardsOptions = data.allRewards;
        let options = rewardsOptions.map((reward) => ({
          label: reward.points,
          value: reward.points,
        }));
        return (
          <div className="mytask">
            <h1 className="rewards__title">
              Rewards for <span>Campaign&nbsp;X</span>
            </h1>
            <Select
              className="rewards__filter"
              onChange={handleSelect}
              options={options}
            ></Select>
            <Task rewards={data}></Task>;
          </div>
        );
      }}
    </Query>
  );
}

export default MyTask;

// quantity: $quantity, points: $points, capPerAmbassador: $capPerAmbassador
