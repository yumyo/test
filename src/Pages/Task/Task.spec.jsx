import React from 'react';
import { shallow } from 'enzyme';
import Task from './Task.jsx';

describe('<Task />', () => {
    let component;
    const props = {
        campaign: {
            title: 'campaign-x-test',
        },
    }

    beforeEach(() => {
        component = shallow(<Task {...props} />);
    });

    it('renders with props', () => {
        expect(component).toMatchSnapshot();
    });
});
