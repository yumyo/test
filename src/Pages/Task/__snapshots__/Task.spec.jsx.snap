// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> renders with props 1`] = `
<h1>
  Empty Task
</h1>
`;
